src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx

```diff
@@ -13,9 +13,39 @@ export default function DeploymentStatusDetailModal({
     environmentName,
     deploymentStatusDetailsBreakdownData,
 }: DeploymentStatusDetailModalType) {
+    const appStatusDetailRef = useRef<HTMLDivElement>(null)
+    const escKeyPressHandler = (evt): void => {
+        if (evt && evt.key === 'Escape' && typeof close === 'function') {
+            evt.preventDefault()
+            close()
+        }
+    }
+    const outsideClickHandler = (evt): void => {
+        if (
+            appStatusDetailRef.current &&
+            !appStatusDetailRef.current.contains(evt.target) &&
+            typeof close === 'function'
+        ) {
+            close()
+        }
+    }
+    useEffect(() => {
+        document.addEventListener('keydown', escKeyPressHandler)
+        return (): void => {
+            document.removeEventListener('keydown', escKeyPressHandler)
+        }
+    }, [escKeyPressHandler])
+
+    useEffect(() => {
+        document.addEventListener('click', outsideClickHandler)
+        return (): void => {
+            document.removeEventListener('click', outsideClickHandler)
+        }
+    }, [outsideClickHandler])
+
     return (
-        <Drawer position="right" width="50%">
-            <div className="deployment-status-breakdown-modal-container bcn-0">
+        <Drawer position="right" width="1024px">
```

Comment:
Hope this strictly providing width will work fine in all screens

Reply:
yes